cln-rpc/src/primitives.rs

```diff
@@ -805,11 +805,16 @@ impl Serialize for OutputDesc {
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Routehop {
-    pub id: PublicKey,
-    pub scid: ShortChannelId,
-    pub feebase: Amount,
-    pub feeprop: u32,
-    pub expirydelta: u16,
+    #[serde(rename = "id")]
+    pub pubkey: PublicKey,
+    #[serde(rename = "scid")]
+    pub short_channel_id: ShortChannelId,
+    #[serde(rename = "feebase")]
+    pub fee_base_msat: Amount,
+    #[serde(rename = "feeprop")]
+    pub fee_proportional_millionths: u32,
+    #[serde(rename = "expirydelta")]
+    pub cltv_expiry_delta: u16,
```

Comment:
Feels like this'd introduce a lot of headaches, such as using one name from these renames in another struct? I'd like to keep the names as in line with the docs as possible. And in this case the need to rename appears to come from a desire to reuse the `Routehop` struct in another location, where these different names are used?

Let's rather create a separate one, and keep them separate, and then start a discussion about renaming them in the schemas themselves, so that all interfacaes remain in sync.

Reply:
Ok i can make them separate, but what about the inconsistency in keysend schema and the original RouteHop primitive?:

keysend schema:
```
"id",
"scid",
"feebase",
"feeprop",
"expirydelta"
```
but then in primitives.proto we have:
```
id = 1;
short_channel_id = 2;
feebase = 3;
feeprop = 4;
expirydelta = 5;
```

for comparison decode/decodepay schema:
```
"pubkey",
"short_channel_id",
"fee_base_msat",
"fee_proportional_millionths",
"cltv_expiry_delta"
```

Reply:
Oh boy, did I go against my own comment again there? I'm sorry, I should not have done that, and we probably should be undoing that pointless name change too :facepalm: You can probably guess that this was a learning experience for me too ^^